src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -25,6 +25,26 @@ public DatabaseSource(DbProviderFactory providerFactory, string connectionString
             CommandText = commandText;
         }
 
+        /// <summary>Creates a new instance of the <see cref="DatabaseSource" /> class.</summary>
+        /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
+        /// <param name="connectionString">The string used to open the connection.</param>
+        /// <param name="commandText">The text command to run against the data source.</param>
+        /// <param name="commandTimeOut">The timeout(in seconds) for database command.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeOut)
```

Comment:
>commandTimeOut [](start = 114, length = 14)

Please check commandTimeOut to make sure it is in a reasonable range.

Reply:
Good catch! as we should let it throw exceptions earlier